conductor-client/src/main/kotlin/com/openlattice/directory/Auth0UserDirectoryService.kt

```diff
@@ -42,28 +42,45 @@ class Auth0UserDirectoryService(
     }
 
     //TODO: Switch over to a Hazelcast map to relieve pressure from Auth0
-    override fun searchAllUsers(searchQuery: String): Map<String, Auth0UserBasic> {
-        logger.info("Searching auth0 users with query: $searchQuery")
+    override fun searchAllUsers(fields: Auth0UserSearchFields): Map<String, User> {
 
-        var page = 0
-        var pageOfUsers = auth0ManagementApi.searchAllUsers(searchQuery, page++, DEFAULT_PAGE_SIZE, "v3")
-        val users = mutableSetOf<Auth0UserBasic>()
-
-        while (pageOfUsers != null) {
-            users.addAll(pageOfUsers)
+        var searchQuery = ""
 
-            if (pageOfUsers.size == DEFAULT_PAGE_SIZE) {
-                pageOfUsers = auth0ManagementApi.searchAllUsers(searchQuery, page++, DEFAULT_PAGE_SIZE, "v3")
-            } else {
-                break
-            }
+        // https://auth0.com/docs/users/user-search/user-search-query-syntax
+        // TODO - support multiple fields and construct a valid Lucene query string to pass to Auth0
+        if (fields.email.orElse("").isNotBlank()) {
+            searchQuery = "email:${fields.email.get()}"
         }
+        else if (fields.name.orElse("").isNotBlank()) {
+            searchQuery = "name:${fields.name.get()}"
+        }
+
+        require(searchQuery.isNotBlank()) { "search query cannot be blank" }
```

Comment:
these checks are unnecessary since blank states are impossible because of constructor assertions. simpler/cleaner to just do 
```
val searchQuery = if (fields.name.isPresent) {
  "name:${fields.name.get()}"
} else {
  "email:${fields.email.get()}"
}
```

(or with null checks and no .get()s if you get rid of the optionals)